### PR TITLE
Raise helpful error when forever is combined with 0 selected tests

### LINKED ISSUE #151: Running no tests forever causes division by zero error
If you run pytest-run-parallel with both the `--forever` option plus no tests selected, you get a pretty strange error.

In the below example, I am using `spin` to run the SciPy test suite, while selecting `--forever` mode, and only running tests which match the pattern `-k asdf`. There are zero tests matching this in SciPy's test suite, so this de-selects all tests.

```
(venv) nodell@b5a8bffbc962:~/scipy-ft$ spin test -s optimize -- -k asdf --forever
Invoking `build` prior to running tests:
$ meson compile -j 6 -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/nodell/scipy-ft/venv/bin/ninja -C /home/nodell/scipy-ft/build -j 6
ninja: Entering directory `/home/nodell/scipy-ft/build'
[2/338] Generating subprojects/highs/src/HConfig.h with a custom command
$ meson install --only-changed -C build --destdir ../build-install --tags=runtime,python-runtime,tests,devel
$ export PYTHONPATH="/home/nodell/scipy-ft/build-install/usr/lib/python3.14t/site-packages"
$ /home/nodell/scipy-ft/venv/bin/python -P -c 'import scipy'
$ cd /home/nodell/scipy-ft/build-install/usr/lib/python3.14t/site-packages
$ /home/nodell/scipy-ft/venv/bin/python -P -m pytest -m 'not slow' -k asdf --forever --pyargs scipy.optimize
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.14.0, pytest-9.0.1, pluggy-1.6.0
rootdir: /home/nodell/scipy-ft
configfile: pytest.ini
plugins: hypothesis-6.148.1, anyio-4.11.0, xdist-3.8.0, cov-7.0.0, timeout-2.4.0, run-parallel-0.7.1
collected 3872 items / 3872 deselected / 0 selected                                                                                                                                                                  
Collected 0 items to run in parallel
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/_pytest/main.py", line 318, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>                          ~~~~^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/_pytest/main.py", line 372, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_hooks.py", line 512, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
INTERNALERROR>            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_callers.py", line 167, in _multicall
INTERNALERROR>     raise exception
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     teardown.throw(exception)
INTERNALERROR>     ~~~~~~~~~~~~~~^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/_pytest/logging.py", line 801, in pytest_runtestloop
INTERNALERROR>     return (yield)  # Run all the tests.
INTERNALERROR>             ^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_callers.py", line 139, in _multicall
INTERNALERROR>     teardown.throw(exception)
INTERNALERROR>     ~~~~~~~~~~~~~~^^^^^^^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/_pytest/terminal.py", line 714, in pytest_runtestloop
INTERNALERROR>     result = yield
INTERNALERROR>              ^^^^^
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pluggy/_callers.py", line 121, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/nodell/scipy-ft/venv/lib/python3.14t/site-packages/pytest_run_parallel/plugin.py", line 212, in pytest_runtestloop
INTERNALERROR>     next_idx = (idx + 1) % number_of_items
INTERNALERROR>                ~~~~~~~~~~^~~~~~~~~~~~~~~~~
INTERNALERROR> ZeroDivisionError: division by zero

============================================================================================== 3872 deselected in 1.56s ==============================================================================================
```

I can think of two ways to approach fixing this.

1. Do what the user literally asked for. Run an infinite loop running no tests. This is not very useful, but it is what the user asked for.
2. Alternatively, if that suggestion doesn't excite you, it would be helpful if the error message were better. For example, it could exit with an error message like "Must have tests selected to use --forever"

### PR DESCRIPTION
**Pull request by lysnikolaou**
Closes #151.

### CODE REVIEW

File: src/pytest_run_parallel/plugin.py [hunk 1]
```diff
@@ -207,6 +207,8 @@ def pytest_runtestloop(self, session: pytest.Session):
             return True
 
         number_of_items = len(session.items)
+        if number_of_items == 0:
+            raise pytest.UsageError("Must have tests selected when passing --forever")
```
Comment (ngoldbaum) [inline src/pytest_run_parallel/plugin.py]: Phrasing tweak and let's add a suggestion for a possible fix: `"Test collection found zero tests when passing --forever. Are you sure you searched for the correct tests?"`

**Comment by ngoldbaum** [comment]
Thanks!
